openstudiocore/src/model/GeneratorMicroTurbineHeatRecovery.hpp

```diff
@@ -0,0 +1,186 @@
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> validHeatRecoveryWaterFlowOperatingModeValues();
+
+  /** @name Getters */
+  //@{
+      
+  unsigned inletPort();
+
+  unsigned outletPort();
+
+  // TODO: Check return type. From object lists, some candidates are: Connection.
+  // boost::optional<Connection> heatRecoveryWaterInletNode() const;
+
+  // TODO: Check return type. From object lists, some candidates are: Connection.
+  // boost::optional<Connection> heatRecoveryWaterOutletNode() const;
+
+  double referenceThermalEfficiencyUsingLowerHeatValue() const;
+  bool isReferenceThermalEfficiencyUsingLowerHeatValueDefaulted() const;
+
+  boost::optional<double> referenceInletWaterTemperature() const;
+
+  std::string heatRecoveryWaterFlowOperatingMode() const;
+
+  bool isHeatRecoveryWaterFlowOperatingModeDefaulted() const;
+
+  boost::optional<double> referenceHeatRecoveryWaterFlowRate() const;
+
+  // TODO: Check return type. From object lists, some candidates are: BiquadraticCurves, BiVariateTables.
+  boost::optional<Curve> heatRecoveryWaterFlowRateFunctionofTemperatureandPowerCurve() const;
+
+  // TODO: Check return type. From object lists, some candidates are: BicubicBiquadraticCurves, BiVariateTables.
+  boost::optional<Curve> thermalEfficiencyFunctionofTemperatureandElevationCurve() const;
+
+  // TODO: Check return type. From object lists, some candidates are: QuadraticCubicCurves, UniVariateTables.
+  boost::optional<Curve> heatRecoveryRateFunctionofPartLoadRatioCurve() const;
+
+  // TODO: Check return type. From object lists, some candidates are: QuadraticCurves, UniVariateTables.
+  boost::optional<Curve> heatRecoveryRateFunctionofInletWaterTemperatureCurve() const;
+
+  // TODO: Check return type. From object lists, some candidates are: QuadraticCurves, UniVariateTables.
+  boost::optional<Curve> heatRecoveryRateFunctionofWaterFlowRateCurve() const;
+
+  double minimumHeatRecoveryWaterFlowRate() const;
+  bool isMinimumHeatRecoveryWaterFlowRateDefaulted() const;
+
+  double maximumHeatRecoveryWaterFlowRate() const;
+  bool isMaximumHeatRecoveryWaterFlowRateDefaulted() const;
+
+  boost::optional<double> maximumHeatRecoveryWaterTemperature() const;
+
+  //@}
+  /** @name Setters */
+  //@{
+
+  // TODO: Check argument type. From object lists, some candidates are: Connection.
+  //bool setHeatRecoveryWaterInletNode(const Connection& connection);
+  //void resetHeatRecoveryWaterInletNode();
+
+  // TODO: Check argument type. From object lists, some candidates are: Connection.
+  //bool setHeatRecoveryWaterOutletNode(const Connection& connection);
+  //void resetHeatRecoveryWaterOutletNode();
+
+  bool setReferenceThermalEfficiencyUsingLowerHeatValue(double referenceThermalEfficiencyUsingLowerHeatValue);
+  void resetReferenceThermalEfficiencyUsingLowerHeatValue();
+
+  void setReferenceInletWaterTemperature(double referenceInletWaterTemperature);
+  void resetReferenceInletWaterTemperature();
```

Comment:
Hum yeah, Technically it's not a required field in E+, it will default to 0, which is dumb with this separation of MicroTurbine and its heat recovery module... Why would you create one to have it at 0 water flow rate...

Goes back to your required withh no default question, how and when do I set them?